### DISCUSSION TITLE
sympy.physics.mechanics

### DISCUSSION THREAD

**Discussion by Peter230655** [sympy/sympy#25089] [Q&A] [answered]
I play around with this quite a bit, doing simulations.
I normally define the time variable like this:
t = sympy.symbols('t')

Recently I saw in a simulation this definition:
t = me.dynamicsymbols._t

I tried it and it worked in my simulations.

For my better understanding of sympy and of sympy.physics.mechanics my question:
What is the difference between the two ways?
Which one is 'better'?

Thanks for any help!!

**Comment by moorepants** [comment, marked as answer]
If you use `dynamicssymbols()` to generate functions of time, it uses the variable for time attached to that function, i.e. `dynamicsymbols._t`. All of the algorithms in physics.vector and physics.mechanics assume the expressions are functions of the symbol `dynamics.symbols._t`. If for example you created `t = symbols('t', real=True)` and tried to create expressions, you'd get invalid results or errors. It happens to be true that `symbols('t') == dynamicsymbols._t` so things work. There is some explanation in the docs here: https://docs.sympy.org/latest/modules/physics/vector/advanced.html?highlight=dynamicsymbols#dynamicsymbols

**Reply by Peter230655** [reply]
Clear! Thanks a lot!
